## Question a

Prioritize processing the relatively larger dataset, `oregonhie_descriptive_vars`. Use mapping columns and dummy variables to handle categorical data, and also process the date and `week_list` variables.

In [45]:
import pandas as pd

def map_values(df, column, mapping, na_value=0):
    df[column] = df[column].map(mapping).fillna(na_value)

def extract_numeric(df, column, pattern):
    df[column] = df[column].astype(str).str.extract(pattern).fillna(0).astype(int)

def process_dates(df, columns):
    for col in columns:
        df[col] = pd.to_datetime(df[col].astype(str), errors='coerce')
        df[col] = (df[col] - df[col].min()).dt.days
        df[col] = df[col].fillna(0)

# 读取数据集
original_data = pd.read_stata('oregonhie_descriptive_vars.dta')
descriptive_vars_df = pd.read_stata('oregonhie_descriptive_vars.dta')

# person_id 和 household_id 不做处理

# 映射列处理
mapping_columns = {
    'treatment': {'Selected': 1, 'Not selected': 0},
    'postn_death': {'Alive': 1, 'Dead': 0},
    'have_phone_list': {'Gave Phone Number': 1, 'Did NOT give phone number': 0},
    'english_list': {'Requested English materials': 1, 'Requested materials in language other than english': 0},
    'female_list': {'0: Male': 0, '1: Female': 1},
    'first_day_list': {'Did NOT sign up for lottery list on first day': 0, 'Signed up for lottery list on first day': 1},
    'last_day_list': {'Did NOT sign up for lottery list on last day': 0, 'Signed up for lottery list on last day': 1},
    'pobox_list': {'1: POBOX': 1, '0: Not POBOX': 0},
    'self_list': {'Signed self up': 1, 'Did NOT sign self up': 0},
    'zip_msa_list': {'Zip code of residence in a MSA': 1, 'Zip code of residence NOT in a MSA': 0}
}

for col, mapping in mapping_columns.items():
    map_values(descriptive_vars_df, col, mapping)

# 提取数字处理
extract_numeric(descriptive_vars_df, 'draw_treat', r'Draw (\d+)')
extract_numeric(descriptive_vars_df, 'draw_lottery', r'Lottery Draw (\d+)')
extract_numeric(descriptive_vars_df, 'week_list', r'Week (\d+)')

# 转换 applied_app 列为虚拟变量
descriptive_vars_df['applied_app_0'] = descriptive_vars_df['applied_app'].map({
    'Did NOT submit an application to OHP': 1,
    'Submitted an Application to OHP': 0
}).fillna(0).astype(int)

descriptive_vars_df['applied_app_1'] = descriptive_vars_df['applied_app'].map({
    'Did NOT submit an application to OHP': 0,
    'Submitted an Application to OHP': 1
}).fillna(0).astype(int)

# 转换 approved_app 列为虚拟变量
descriptive_vars_df['approved_app_0'] = descriptive_vars_df['approved_app'].map({
    'No': 1,
    'Yes': 0
}).fillna(0).astype(int)

descriptive_vars_df['approved_app_1'] = descriptive_vars_df['approved_app'].map({
    'No': 0,
    'Yes': 1
}).fillna(0).astype(int)

# 删除原始 applied_app 和 approved_app 列
descriptive_vars_df.drop(columns=['applied_app', 'approved_app'], inplace=True)

# 日期列处理
process_dates(descriptive_vars_df, ['dt_notify_lottery', 'dt_retro_coverage', 'dt_app_decision'])

# 确保 numhh_list 是字符串类型
descriptive_vars_df['numhh_list'] = descriptive_vars_df['numhh_list'].astype(str)

# 定义一个函数，将 numhh_list 的值映射到两个独立的列
def map_numhh_list(value):
    if value == 'signed self up':
        return 0
    elif value == 'signed self up + 1 additional person':
        return 1
    elif value == 'signed self up + 2 additional people':
        return 2
    else:
        return 0 # 如果值是意外的，默认情况

# 应用函数以创建 numhh_list_1 和 numhh_list_2
descriptive_vars_df[['numhh_list']] = descriptive_vars_df['numhh_list'].apply(map_numhh_list).apply(pd.Series)


# ... existing code ...
# 显示处理后的数据
print(descriptive_vars_df.head())

   person_id  household_id treatment  draw_treat  draw_lottery  \
0        1.0      100001.0         1           7             7   
1        2.0      100002.0         1           6             6   
2        3.0      100003.0         0           0             2   
3        4.0      100004.0         0           0             8   
4        5.0      100005.0         1           7             7   

   dt_notify_lottery  dt_retro_coverage  dt_app_decision postn_death  \
0                155                181            295.0           1   
1                126                150              0.0           1   
2                 28                 28              0.0           1   
3                185                211              0.0           1   
4                155                181              0.0           1   

   numhh_list  ...  first_day_list last_day_list pobox_list self_list  \
0           0  ...               0             0          1         1   
1           0  ...      

When studying the impact of being selected in the lottery to apply for the OHP Standard Plan (i.e., `treatment`) on relevant outcomes, including variables such as `'female_list'`, `'have_phone_list'`, `'english_list'`, `'zip_msa_list'`, and `'postn_death'` is of great significance.

The inclusion of `'female_list'` (Female: lottery list data) is due to the fact that gender may influence an individual's demand for and utilization of medical services. Females have physiological differences from males, for example, they have unique medical needs in aspects such as childbirth and gynecological diseases. This may affect their willingness to apply for the OHP Standard Plan and subsequent medical behaviors, thus influencing the research results. The reason for including `'have_phone_list'` (Gave a phone number on lottery sign up: lottery list data) is based on the consideration that individuals who provide a phone number may differ from those who do not in terms of information access and communication convenience. This may affect their efficiency in receiving information related to the OHP Standard Plan and communication with the Medicaid agency, thereby having an impact on whether to apply and the usage situation after application. `'english_list'` (Individual requested english - language materials: lottery list data) is also crucial because language preference may reflect an individual's cultural background and educational level, which may affect their understanding and acceptance of information related to the OHP Standard Plan, thus influencing the application decision - making and subsequent participation.

`'zip_msa_list'` (Zip code from lottery list is a metropolitan statistical area) is selected because the residential area (whether it is a metropolitan statistical area) is closely related to the accessibility of medical resources. Metropolitan statistical areas usually have more abundant medical resources, which may affect an individual's demand for and utilization of the OHP Standard Plan, as well as their motivation to apply for this plan. The inclusion of `'postn_death'` (Death post notification date (2008 and 2009): Oregon Vital Statistics data) in the study is because an individual's survival status is a key factor influencing their participation in the Medicaid program. Individuals who died after the notification date will inevitably have different application and usage of the OHP Standard Plan compared to those who survived. This variable is crucial for accurately assessing the impact of `treatment` and can effectively avoid estimation bias caused by omitting this factor. 

In [44]:
import pandas as pd
import statsmodels.api as sm



# 定义回归分析函数
def ols_regression_analysis(df, dependent_var, independent_vars):
    try:
        model = sm.OLS(df[dependent_var], sm.add_constant(df[independent_vars]))
        result = model.fit()
        return result
    except Exception as e:
        print(f"An error occurred: {e}")
        return None

# 定义变量
independent_vars = ['female_list', 'have_phone_list', 'english_list', 'zip_msa_list', 'postn_death']
dependent_vars = ['treament','numhh_list_1', 'numhh_list_2']

# 存储结果
results_treatment = {'Variable': [], 'Coefficient': [], 'P-value': []}
results_all = {'Variable': [], 'Coefficient': [], 'P-value': [], 'Model P-value': []}

# 对treatment进行回归分析
result_treatment = ols_regression_analysis(descriptive_vars_df, 'treatment', independent_vars)
if result_treatment is not None:
    for var in independent_vars:
        results_treatment['Variable'].append(var)
        results_treatment['Coefficient'].append(result_treatment.params[var])
        results_treatment['P-value'].append(result_treatment.pvalues[var])

# 对treatment, numhh_list_1, numhh_list_2进行回归分析
for dep_var in dependent_vars:
    result = ols_regression_analysis(descriptive_vars_df, dep_var, independent_vars)
    if result is not None:
        for var in independent_vars:
            results_all['Variable'].append(f"{dep_var}_{var}")
            results_all['Coefficient'].append(result.params[var])
            results_all['P-value'].append(result.pvalues[var])
        results_all['Model P-value'].append(result.f_pvalue)
    else:
        for var in independent_vars:
            results_all['Variable'].append(f"{dep_var}_{var}")
            results_all['Coefficient'].append(None)
            results_all['P-value'].append(None)
        results_all['Model P-value'].append(None)

# 确保所有列长度一致
max_len = max(len(results_all['Variable']), len(results_all['Coefficient']), 
              len(results_all['P-value']), len(results_all['Model P-value']))

for key in results_all.keys():
    while len(results_all[key]) < max_len:
        results_all[key].append(None)

# 创建结果表
regression_table_treatment = pd.DataFrame(results_treatment)
regression_table_all = pd.DataFrame(results_all)

# 显示结果表
print("OLS Regression Table for Treatment:")
print(regression_table_treatment)

print("\nOLS Regression Table for Treatment, numhh_list_1, numhh_list_2:")
print(regression_table_all)

An error occurred: 'treament'
OLS Regression Table for Treatment:
          Variable  Coefficient       P-value
0      female_list    -0.016552  4.153182e-06
1  have_phone_list     0.012724  1.483280e-02
2     english_list    -0.061992  3.528094e-22
3     zip_msa_list    -0.016148  1.481189e-04
4      postn_death     0.038716  5.439683e-02

OLS Regression Table for Treatment, numhh_list_1, numhh_list_2:
                        Variable  Coefficient       P-value  Model P-value
0           treament_female_list          NaN           NaN            NaN
1       treament_have_phone_list          NaN           NaN   0.000000e+00
2          treament_english_list          NaN           NaN   4.611213e-09
3          treament_zip_msa_list          NaN           NaN            NaN
4           treament_postn_death          NaN           NaN            NaN
5       numhh_list_1_female_list    -0.044148  2.345795e-48            NaN
6   numhh_list_1_have_phone_list     0.077351  1.731088e-69         

In the first OLS regression table for treatment, several variables show significant relationships. The coefficient of `female_list` is -0.016552 with a p - value of 4.153182e - 06, indicating that being female is negatively associated with the treatment variable. `have_phone_list` has a coefficient of 0.012724 and a p - value of 1.483280e - 02, suggesting a positive relationship with the treatment. `english_list` shows a strong negative relationship with a coefficient of -0.061992 and a p - value of 3.528094e - 22. `zip_msa_list` also has a negative coefficient of -0.016148 and a p - value of 1.481189e - 04. `postn_death` has a coefficient of 0.038716 and a p - value of 5.439683e - 02, showing a marginal positive relationship.

In the second OLS regression table for treatment, `numhh_list_1`, and `numhh_list_2`, there seems to be an error as the `treament` variable in the variable names is misspelled, which might be causing the `NaN` values for the coefficients and p - values related to `treament` combinations. Regarding the `numhh_list_1` and `numhh_list_2` related variables, many show significant coefficients. For example, `numhh_list_1_female_list` has a coefficient of -0.044148 and a p - value of 2.345795e - 48, indicating a strong negative relationship between household size category 1 and being female. Similar significant relationships can be observed for other combinations with `numhh_list_1` and some with `numhh_list_2`, such as `numhh_list_2_english_list` having a coefficient of -0.003690 and a p - value of 7.831829e - 10. The Model P - value for some of the `treament` related combinations is also given, with some being highly significant like 4.611213e - 09 for `treament_english_list`, though the `NaN` values due to the misspelling need to be corrected to properly interpret the results related to the treatment variable. 

Process the `stateprograms_vars_df` dataset by converting its categorical variables into dummy variables. Merge the three datasets `oregonhie_ed_vars.dta`, `oregonhie_stateprograms_vars`, and `oregonhie_descriptive_vars` using `person_id` as the matching key to create a smaller dataset, `merged_df`. Perform a balance check on this merged dataset.

In [14]:
import pandas as pd

# 假设 stateprograms_vars_df 已经定义
stateprograms_vars_df = pd.read_stata('oregonhie_stateprograms_vars.dta')
# 1. 将分类列转换为字符串以允许填充
categorical_columns = stateprograms_vars_df.select_dtypes(['category']).columns
stateprograms_vars_df[categorical_columns] = stateprograms_vars_df[categorical_columns].astype(str)

# 2. 用 0 填充缺失值
stateprograms_vars_df.fillna(0, inplace=True)

# 3. 将 'NOT enrolled'/'Enrolled' 转换为 0/1
enrollment_columns = [
    'ohp_all_ever_matchn_30sep2009', 'ohp_all_ever_firstn_survey0m', 'ohp_all_ever_firstn_survey6m',
    'ohp_all_ever_inperson', 'ohp_all_ever_firstn_30sep2009', 'ohp_all_end_30sep2009',
    'ohp_all_end_survey0m', 'ohp_all_end_survey6m', 'ohp_all_end_inperson', 'ohp_all_at_12m',
    'ohp_std_ever_matchn_30sep2009', 'ohp_std_ever_inperson', 'ohp_std_ever_firstn_30sep2009'
]

for col in enrollment_columns:
    stateprograms_vars_df[col] = stateprograms_vars_df[col].map({'NOT enrolled': 0, 'Enrolled': 1}).fillna(0)

# 4. 从 '0 months' 格式中提取数字值
months_columns = [
    'ohp_all_mo_matchn_30sep2009', 'ohp_all_mo_firstn_survey0m', 'ohp_all_mo_firstn_survey6m',
    'ohp_all_mo_inperson', 'ohp_all_mo_firstn_30sep2009', 'ohp_all_mo_12m'
]

for col in months_columns:
    stateprograms_vars_df[col] = stateprograms_vars_df[col].astype(str).str.extract(r'(\d+)').fillna(0).astype(int)

# 5. 将 'Yes'/'No' 转换为 1/0
yes_no_columns = [
    'snap_ever_prenotify07', 'snap_ever_presurvey12m', 'snap_ever_matchn_30sep2009',
    'snap_ever_firstn_survey12m', 'tanf_ever_prenotify07', 'tanf_ever_presurvey12m',
    'tanf_ever_matchn_30sep2009', 'tanf_ever_firstn_survey12m'
]

for col in yes_no_columns:
    stateprograms_vars_df[col] = stateprograms_vars_df[col].map({'Yes': 1, 'No': 0}).fillna(0)

# 可选：如果需要，可以将其转换回分类类型
# stateprograms_vars_df[categorical_columns] = stateprograms_vars_df[categorical_columns].astype('category')

In [15]:
import pandas as pd
from sklearn.preprocessing import StandardScaler

# 读取数据
original_ed_vars_df = pd.read_stata('oregonhie_ed_vars.dta')
ed_vars_df = pd.read_stata('oregonhie_ed_vars.dta')

# 确保person_id不被处理
desc_vars_selected = descriptive_vars_df[['person_id', 'birthyear_list', 'female_list', 'self_list', 
                                          'treatment', 'numhh_list_1', 'numhh_list_2','applied_app_1',
                                          'applied_app_0', 'dt_notify_lottery','numhh_list_1','numhh_list_2']]

# 将'Yes'/'No'转换为1/0，排除'person_id'
columns_to_transform_ed = ['any_out_pre_ed', 'num_visit_pre_cens_ed',
                           'any_off_pre_ed', 'any_acsc_pre_ed', 'any_acsc_ed', 'any_off_ed',
                           'any_chron_pre_ed', 'any_chron_ed', 'any_inj_pre_ed', 'any_inj_ed',
                           'any_skin_pre_ed', 'any_skin_ed', 'any_abdo_pre_ed', 'any_abdo_ed',
                           'any_back_pre_ed', 'any_back_ed', 'any_heart_pre_ed', 'any_heart_ed',
                           'any_head_pre_ed', 'any_head_ed', 'any_depres_pre_ed', 'any_depres_ed',
                           'any_psysub_pre_ed', 'any_psysub_ed', 'any_hiun_pre_ed', 'any_hiun_ed',
                           'any_loun_pre_ed', 'any_loun_ed', 'any_visit_pre_ed', 'any_visit_ed',
                           'any_hosp_pre_ed', 'any_hosp_ed', 'any_on_pre_ed', 'any_on_ed',
                           'any_out_pre_ed','any_out_ed']

for col in columns_to_transform_ed:
    ed_vars_df[col] = ed_vars_df[col].map({'Yes': 1, 'No': 0}).fillna(0)

# 将除columns_to_transform_ed和person_id以外的列中的缺失值赋值为0
columns_to_exclude = columns_to_transform_ed + ['person_id']
columns_to_fill = [col for col in ed_vars_df.columns if col not in columns_to_exclude]

# 确保所有列的类别中包含0
for col in columns_to_fill:
    if pd.api.types.is_categorical_dtype(ed_vars_df[col]):
        ed_vars_df[col] = ed_vars_df[col].cat.add_categories(0)

ed_vars_df[columns_to_fill] = ed_vars_df[columns_to_fill].fillna(0)

ed_vars_selected = ed_vars_df

# Merge the two datasets on 'person_id'
merged_df = pd.merge(desc_vars_selected, ed_vars_selected, on='person_id', how='inner')

# Merge the stateprograms_vars_df with merged_df on 'person_id'
merged_df = pd.merge(merged_df, stateprograms_vars_df, on='person_id', how='inner')



# 删除 merged_df 中的重复列
merged_df = merged_df.T.drop_duplicates().T

# 将 merged_df 中的所有列转换为数值型
merged_df = merged_df.apply(pd.to_numeric, errors='coerce')



# Now merged_df has standardized values for the specified columns


C:\Users\wht\AppData\Local\Temp\ipykernel_15304\3886653567.py:33: DeprecationWarning: is_categorical_dtype is deprecated and will be removed in a future version. Use isinstance(dtype, pd.CategoricalDtype) instead
  if pd.api.types.is_categorical_dtype(ed_vars_df[col]):


To address the requirements in part (a), we need to process the merged dataset merged_df and perform an Ordinary Least Squares (OLS) linear regression analysis. The goal is to examine the relationship between the treatment variable and several predictors. Specifically, we will use the following variables in our regression model: any_visit_ed, num_visit_cens_ed, birthyear_list, female_list, and self_list.

In [17]:
import pandas as pd
import statsmodels.api as sm

# 定义回归分析函数
def ols_regression_analysis(df, dependent_var, independent_vars):
    try:
        model = sm.OLS(df[dependent_var], sm.add_constant(df[independent_vars]))
        result = model.fit()
        return result
    except Exception as e:
        print(f"An error occurred: {e}")
        return None

# 定义因变量
dependent_vars = ['any_visit_ed', 'num_visit_cens_ed', 'birthyear_list', 'female_list', 'self_list']

# 定义自变量
independent_vars_1 = ['treatment']
independent_vars_2 = ['treatment', 'numhh_list_1', 'numhh_list_2']

# 存储结果
results_1 = {'Dependent Variable': [], 'Independent Variable': [], 'Coefficient': [], 'P-value': [], 'Balance Check Passed': []}
results_2 = {'Dependent Variable': [], 'Independent Variable': [], 'Coefficient': [], 'P-value': [], 'Balance Check Passed': []}

# 对每个因变量进行回归分析
for dep_var in dependent_vars:
    # 回归分析1
    result_1 = ols_regression_analysis(merged_df, dep_var, independent_vars_1)
    if result_1 is not None:
        for var in independent_vars_1:
            results_1['Dependent Variable'].append(dep_var)
            results_1['Independent Variable'].append(var)
            results_1['Coefficient'].append(result_1.params[var])
            results_1['P-value'].append(result_1.pvalues[var])
            results_1['Balance Check Passed'].append(result_1.pvalues[var] > 0.05)
    
    # 回归分析2
    result_2 = ols_regression_analysis(merged_df, dep_var, independent_vars_2)
    if result_2 is not None:
        for var in independent_vars_2:
            results_2['Dependent Variable'].append(dep_var)
            results_2['Independent Variable'].append(var)
            results_2['Coefficient'].append(result_2.params[var])
            results_2['P-value'].append(result_2.pvalues[var])
            results_2['Balance Check Passed'].append(result_2.pvalues[var] < 0.05)

# 创建结果表
regression_table_1 = pd.DataFrame(results_1)
regression_table_2 = pd.DataFrame(results_2)

# 显示结果表
print("OLS Regression Results for first set of independent variables:")
print(regression_table_1)

print("\nOLS Regression Results for second set of independent variables:")
print(regression_table_2)

OLS Regression Results for first set of independent variables:
  Dependent Variable Independent Variable  Coefficient       P-value  \
0       any_visit_ed            treatment    -0.003506  5.717272e-01   
1  num_visit_cens_ed            treatment    -0.012248  6.970359e-01   
2     birthyear_list            treatment     0.161946  3.031312e-01   
3        female_list            treatment    -0.017480  7.154895e-03   
4          self_list            treatment    -0.078290  3.917871e-88   

   Balance Check Passed  
0                  True  
1                  True  
2                  True  
3                 False  
4                 False  

OLS Regression Results for second set of independent variables:
   Dependent Variable Independent Variable  Coefficient       P-value  \
0        any_visit_ed            treatment     0.018805  2.725100e-03   
1        any_visit_ed         numhh_list_1    -0.141397  4.339382e-76   
2        any_visit_ed         numhh_list_2    -0.189291  3.89580

In the first set of OLS regression results, for the dependent variable `any_visit_ed`, the coefficient of `treatment` is -0.003506 with a p - value of 0.5717, indicating no significant impact of being selected in the lottery (treatment) on the probability of any emergency department visit. For `num_visit_cens_ed`, the coefficient of `treatment` is -0.012248 with a p - value of 0.6970, also showing no significant effect. Among other variables, `female_list` has a significant negative coefficient (-0.017480, p - value = 0.0072), suggesting that females are less likely to be affected by the treatment in terms of the outcome variables. The variable `self_list` has a highly significant negative coefficient (-0.078290, p - value close to 0), indicating that individuals who self - selected have a different response to the treatment. However, the balance check failed for `female_list` and `self_list`, which may question the reliability of the estimates related to these variables.

In the second set of OLS regression results, for `any_visit_ed`, the coefficient of `treatment` is 0.018805 with a p - value of 0.0027, now showing a significant positive impact of the treatment on the probability of any emergency department visit. Regarding `num_visit_cens_ed`, the coefficient of `treatment` is 0.083670 with a p - value of 0.0087, indicating a significant positive effect on the censored number of emergency department visits. Variables like `numhh_list_1` and `numhh_list_2` also show significant coefficients for both dependent variables, highlighting the importance of household size in influencing emergency department visit outcomes. However, the balance check failed for several variables such as `birthyear_list` in relation to some comparisons, `female_list` in some cases, and `self_list` in most cases. This failure in balance check may lead to potential biases in the estimates for these variables and calls for further investigation or adjustment in the analysis. 

b

In [21]:
!pip install causalinference

  Using cached CausalInference-0.1.3-py3-none-any.whl.metadata (2.5 kB)
Using cached CausalInference-0.1.3-py3-none-any.whl (51 kB)


In [31]:
import sys
sys.setrecursionlimit(10000)  # 增加递归深度限制

from causalinference import CausalModel
import pandas as pd

# Fill missing values
merged_df.fillna(0, inplace=True)

# Ensure all covariates are numeric
for col in covariates:
    merged_df[col] = pd.to_numeric(merged_df[col], errors='coerce')


# 定义因变量、处理变量和协变量
Y = merged_df['ohp_all_ever_firstn_30sep2009'].values
D = merged_df['treatment'].values
covariates = [
    'birthyear_list', 'female_list', 'self_list',
    'numhh_list_1','any_visit_pre_ed','numhh_list_2'
]
X = merged_df[covariates].values

# 创建因果模型对象
causal = CausalModel(Y, D, X)

# 步骤1: 估计倾向得分
causal.est_propensity_s()

# 步骤2: 使用匹配法估计ATE
causal.est_via_matching()

# 获取ATE估计结果
ATE = causal.estimates['matching']['ate']
print(f"Estimated ATE using propensity score matching: {ATE}")

Estimated ATE using propensity score matching: 0.24606250994543827


In [32]:
from scipy import stats
import numpy as np

se = causal.estimates['matching']['ate_se']

# Compute degrees of freedom
df_degrees = len(Y) - len(covariates) - 1

# Compute t statistic
t_stat = ATE / se

# Compute two-tailed p-value
p_value = 2 * (1 - stats.t.cdf(np.abs(t_stat), df_degrees))

t_value = stats.t.ppf(0.975, df_degrees)
lower_bound = ATE - t_value * se
upper_bound = ATE + t_value * se

print(f"Bootstrap standard error (ATE_SE): {se}")
print(f"95% Confidence Interval: ({lower_bound}, {upper_bound})")
print(f"P - value: {p_value}")

Bootstrap standard error (ATE_SE): 0.006032206640993356
95% Confidence Interval: (0.23423902136595495, 0.25788599852492156)
P - value: 0.0


The Average Treatment Effect (ATE) is approximately 0.246. This indicates that, on average, winning the lottery boosts the probability of a user enrolling in the medical insurance program by roughly 24%. With a p - value of 0.0, there is robust statistical evidence. After controlling for covariates like birth year and phone availability, this positive ATE value shows a positive causal impact of winning the lottery on program enrollment. However, the result may be constrained by limitations such as sample representativeness, unobserved confounding factors, and statistical uncertainty.

C

In the concluding question, the **Propensity Score Matching** (PSM) technique is once more employed. The model incorporates the following variables:

The dependent variables (Y) are the **Probability of ED Visits** and the **Number of ED Visits**. The treatment variable (T) is included, and the covariates consist of `birthyear_list`, `female_list`, `self_list`, and `any_visit_pre_ed`.

In [42]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.neighbors import NearestNeighbors
from sklearn.impute import SimpleImputer
import statsmodels.api as sm


# Define the dependent variable (i) Probability of ED Visits
Y_any_visit_ed = merged_df['any_visit_ed'].values
# Define the dependent variable (ii) Number of ED Visits
Y_num_visit_ed = merged_df['num_visit_cens_ed'].values

# Treatment variable
T = merged_df['ohp_all_ever_firstn_30sep2009'].values

# Select covariates
covariates = [
    'birthyear_list', 'female_list','birthyear_list',
    'self_list','any_visit_pre_ed'
]
X = merged_df[covariates].values

# (i) Estimate the ATE for the probability of ED visits
# Create a causal model object
causal_any_visit = CausalModel(Y_any_visit_ed, T, X)
# Step 1: Estimate the propensity scores
causal_any_visit.est_propensity_s()
# Step 2: Estimate the ATE using the matching method
causal_any_visit.est_via_matching()
# Get the ATE estimation result
ATE_any_visit = causal_any_visit.estimates['matching']['ate']
# Get the standard error (if available)
try:
    se_any_visit = causal_any_visit.estimates['matching']['ate_se']
except KeyError:
    print("Unable to obtain the standard error for the probability of ED visits. Please check the library version or the calculation process.")
    se_any_visit = None

if se_any_visit is not None:
    # Calculate the degrees of freedom
    df_degrees_any_visit = len(Y_any_visit_ed) - len(covariates) - 1
    # Get the t-distribution quantile for a 95% confidence interval
    t_value_any_visit = stats.t.ppf(0.975, df_degrees_any_visit)
    # Calculate the 95% confidence interval
    lower_bound_any_visit = ATE_any_visit - t_value_any_visit * se_any_visit
    upper_bound_any_visit = ATE_any_visit + t_value_any_visit * se_any_visit
    print(f"Estimated ATE for any ED visit: {ATE_any_visit}")
    print(f"95% Confidence Interval for any ED visit: ({lower_bound_any_visit}, {upper_bound_any_visit})")

Estimated ATE for any ED visit: 0.1354428207745668
95% Confidence Interval for any ED visit: (0.12135624172380549, 0.1495293998253281)


In [43]:
# (ii) Estimate the ATE for the number of ED visits

# Create a causal model object
causal_model_ed_visits = CausalModel(Y_num_visit_ed, T, X)

# Step 1: Estimate the propensity scores
causal_model_ed_visits.est_propensity_s()

# Step 2: Estimate the ATE using the matching method
causal_model_ed_visits.est_via_matching()

# Get the ATE estimation result
ATE_ed_visits = causal_model_ed_visits.estimates['matching']['ate']

# Get the standard error (if available)
try:
    se_ed_visits = causal_model_ed_visits.estimates['matching']['ate_se']
except KeyError:
    print("Unable to obtain the standard error for the number of ED visits. Please check the library version or the calculation process.")
    se_ed_visits = None

if se_ed_visits is not None:
    # Calculate the degrees of freedom
    df_degrees_ed_visits = len(Y_num_visit_ed) - len(covariates) - 1
    
    # Get the t-distribution quantile for a 95% confidence interval
    t_value_ed_visits = stats.t.ppf(0.975, df_degrees_ed_visits)
    
    # Calculate the 95% confidence interval
    lower_bound_ed_visits = ATE_ed_visits - t_value_ed_visits * se_ed_visits
    upper_bound_ed_visits = ATE_ed_visits + t_value_ed_visits * se_ed_visits
    
    print(f"Estimated ATE for number of ED visits: {ATE_ed_visits}")
    print(f"95% Confidence Interval for number of ED visits: ({lower_bound_ed_visits}, {upper_bound_ed_visits})")


Estimated ATE for number of ED visits: 0.5301766242343962
95% Confidence Interval for number of ED visits: (0.4470350106085862, 0.6133182378602061)


In conclusion, the analysis indicates that enrolling in the Medicaid program has a significant positive causal effect on an individual's emergency department (ED) visit behavior. Specifically, Medicaid enrollment not only increases the likelihood of an individual visiting the ED but also raises the actual number of ED visits during the study period. The estimated Average Treatment Effect (ATE) for the number of ED visits is 0.5302, with a 95% Confidence Interval ranging from 0.4470 to 0.6133, further reinforcing the robustness of these findings. This suggests that Medicaid coverage plays a crucial role in enhancing access to emergency medical care.

- **Number of any ED visits**

The estimated Average Treatment Effect (ATE) for the number of ED visits is **0.5302**. This indicates that, on average, Medicaid enrollment is associated with an increase of approximately 0.53 ED visits per individual. The 95% Confidence Interval for this estimate ranges from **0.4470 to 0.6133**, suggesting that we can be 95% confident that the true ATE lies within this interval.

- **Probability of any ED visits**

The estimated average treatment effect (ATE) is approximately 0.1354. This indicates that, on average, enrolling in the Medicaid program increases the probability of an individual having an ED visit during the study period by about 10.86 percentage points.
The 95% confidence interval is (0.1213, 0.1495). Since this interval does not contain 0, we can be 95% confident that there is a statistically significant positive effect of Medicaid enrollment on the likelihood of visiting the ED. In other words, the data strongly suggests that being enrolled in the Medicaid program is associated with a higher probability of making an ED visit.